source/served/ddoc.d

```diff
@@ -3,14 +3,101 @@ module served.ddoc;
 import served.protocol;
 
 import std.string;
+import std.format;
 import ddoc;
 
+
+/**
+ * A test function for checking `DDoc` parsing
+ *
+ * Params:
+ *     hello = a string
+ *     world = an integer
+ *
+ * Author: Jonny
+ * Bugs: None
+ * ---
+ * import std.stdio;
+ *
+ * int main(string[] args) {
+ * 	   writeln("Testing inline code")
+ * }
+ * ---
+ */
+private int testFunction(string foo, int bar)
+{
+	import std.stdio : writeln;
+	writeln(foo, bar);
+	return 0;
+}
+
+
+/**
+ * Convert a Ddoc comment string to markdown. Returns ddoc string back if it is
+ * not valid.
+ * Params:
+ *		ddoc = string of a valid Ddoc comment.
+ */
 string ddocToMarkdown(string ddoc)
 {
-	auto lexer = Lexer(prepareDDoc(ddoc), true);
-	return expand(lexer, markdownMacros, false).replace("&#36;", "$");
+	// Parse ddoc. Return if exception.
+	Comment comment;
+	try
+	{
+		comment = parseComment(prepareDDoc(ddoc), markdownMacros);
+	}
+	catch (Exception e)
+	{
+		return ddoc;
```

Comment:
Could you fall back to returning the expanded & prepared ddoc macros like before if it fails to parse?

Reply:
I was a bit concerned about the Lexer throwing an exception in prepare Ddoc so I left it out. What do you think?

Reply:
Have you tried if it can throw with any invalid input? All it's doing is tokenizing and not interpreting what it means so I don't think that it will throw anything

Reply:
Yeah just tried. `prepareDDoc("\n\n汉语\n---");` will throw an exception.